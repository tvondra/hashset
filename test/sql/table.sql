CREATE TABLE users (
    user_id int PRIMARY KEY,
    user_likes int4hashset DEFAULT int4hashset(capacity := 2)
);
INSERT INTO users (user_id) VALUES (1);
UPDATE users SET user_likes = hashset_add(user_likes, 101) WHERE user_id = 1;
UPDATE users SET user_likes = hashset_add(user_likes, 202) WHERE user_id = 1;
SELECT hashset_contains(user_likes, 101) FROM users WHERE user_id = 1;
SELECT hashset_count(user_likes) FROM users WHERE user_id = 1;
SELECT hashset_sorted(user_likes) FROM users WHERE user_id = 1;
