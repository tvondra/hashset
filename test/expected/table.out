CREATE TABLE users (
    user_id int PRIMARY KEY,
    user_likes hashset DEFAULT hashset_init(2)
);
INSERT INTO users (user_id) VALUES (1);
UPDATE users SET user_likes = hashset_add(user_likes, 101) WHERE user_id = 1;
UPDATE users SET user_likes = hashset_add(user_likes, 202) WHERE user_id = 1;
SELECT hashset_contains(user_likes, 101) FROM users WHERE user_id = 1;
 hashset_contains 
------------------
 t
(1 row)

SELECT hashset_count(user_likes) FROM users WHERE user_id = 1;
 hashset_count 
---------------
             2
(1 row)

SELECT hashset_sorted(user_likes) FROM users WHERE user_id = 1;
 hashset_sorted 
----------------
 {101,202}
(1 row)

